kernel/src/main/java/org/kframework/unparser/KPrint.java

```diff
@@ -128,7 +137,17 @@ public void prettyPrint(Definition def, Module module, Consumer<byte[]> print, K
                 RuleGrammarGenerator gen = new RuleGrammarGenerator(def);
                 Module unparsingModule = RuleGrammarGenerator.getCombinedGrammar(gen.getProgramsGrammar(module), false).getParsingModule();
                 return (unparseTerm(result, unparsingModule, colorize) + "\n").getBytes();
-            } default:
+            }
+            case KORE:
+                if (compiledDefinition.isPresent()) {
+                    CompiledDefinition cdef = compiledDefinition.get();
+                    ModuleToKORE converter = new ModuleToKORE(module, files, cdef.topCellInitializer, kompileOptions);
+                    K result = ExpandMacros.forNonSentences(module, files, kompileOptions, false).expand(orig);
+                    result = new AddSortInjections(module).addInjections(result);
+                    converter.convert(result);
+                    return converter.toString().getBytes();
+                }
```

Comment:
We should have an `else` branch which throws a `KEMException.criticalError` so that devs can catch issues in invocations of this easier/earlier.

Reply:
I thought that was unnecessary because if the condition doesn't hold it will fall through to the default case which throws an exception.

However, I agree this make the program logic harder, therefore I added another throw locally.

Reply:
I wasn't aware that Java will cause a fall-through if the condition doesn't hold. If it's the case maybe we don't need the extra branch then, but I can make that refactor in a follow up PR.